src/main/java/org/testng/TestRunner.java

```diff
@@ -678,7 +680,6 @@ public int compareTo(@Nonnull IWorker<ITestNGMethod> other) {
         });
 
     runJUnitWorkers(workers);
-    m_allTestMethods = runMethods.toArray(new ITestNGMethod[0]);
```

Comment:
If you remove this line, you'll have to remove `runMethods` too.
I'm not feeling it good because I can't imagine it is useless.
It looks the `allTestMethods` computation is different in the case of JUnit.